sql/core/src/main/scala/org/apache/spark/sql/execution/basicOperators.scala

```diff
@@ -39,15 +39,26 @@ case class Project(projectList: Seq[NamedExpression], child: SparkPlan)
     child.asInstanceOf[CodegenSupport].produce(ctx, this)
   }
 
+  override def usedInputs: AttributeSet = {
```

Comment:
As discussed offline, this is need to handle the cases that an Attribute could appear twice in `projectList`, but we can't generate the code for an Attribute twice.